seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSource.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.mongodb.source;
+
+import static org.apache.seatunnel.connectors.seatunnel.mongodb.config.Config.COLLECTION;
+import static org.apache.seatunnel.connectors.seatunnel.mongodb.config.Config.DATABASE;
+import static org.apache.seatunnel.connectors.seatunnel.mongodb.config.Config.SCHEMA;
+import static org.apache.seatunnel.connectors.seatunnel.mongodb.config.Config.URI;
+
+import org.apache.seatunnel.api.common.PrepareFailException;
+import org.apache.seatunnel.api.source.Boundedness;
+import org.apache.seatunnel.api.source.SeaTunnelSource;
+import org.apache.seatunnel.api.source.SourceReader;
+import org.apache.seatunnel.api.source.SourceSplitEnumerator;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.common.config.CheckConfigUtil;
+import org.apache.seatunnel.common.config.CheckResult;
+import org.apache.seatunnel.common.config.TypesafeConfigUtils;
+import org.apache.seatunnel.common.constants.PluginType;
+import org.apache.seatunnel.common.utils.JsonUtils;
+import org.apache.seatunnel.connectors.seatunnel.mongodb.state.MongodbSourceState;
+import org.apache.seatunnel.connectors.seatunnel.mongodb.utils.RawTypeConverter;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigBeanFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.auto.service.AutoService;
+
+import java.util.Iterator;
+import java.util.Map;
+
+@AutoService(SeaTunnelSource.class)
+public class MongodbSource implements SeaTunnelSource<SeaTunnelRow, MongodbSourceSplit, MongodbSourceState> {
+
+    private SeaTunnelRowType rowTypeInfo;
+
+    private MongodbSourceEvent params;
+
+    String confPrefix = "readconfig.";
+
+    @Override
+    public String getPluginName() {
+        return "MongoDB";
+    }
+
+    @Override
+    public void prepare(Config config) throws PrepareFailException {
+        Config extractConfig = TypesafeConfigUtils.extractSubConfig(config, confPrefix, false);
+        this.params = ConfigBeanFactory.create(extractConfig, MongodbSourceEvent.class);
+
+        CheckResult result = CheckConfigUtil.checkAllExists(extractConfig, URI, DATABASE, COLLECTION);
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
+        }
+
+        if (extractConfig.hasPath(SCHEMA)) {
```

Comment:
Now setting shema information in connector v2 we have a unified approach, you can refer to https://github.com/apache/incubator-seatunnel/pull/2439/files#diff-7c704c6a3927123b31b41483102f286339da3832309810496f19410c2548f06f.